confd/tests/test_suite_common.sh

```diff
@@ -927,3 +931,168 @@ EOF
     $CALICOCTL delete node node1
     $CALICOCTL delete node node2
 }
+
+test_node_mesh_bgp_password() {
+    # For KDD, run Typha and clean up the output directory.
+    if [ "$DATASTORE_TYPE" = kubernetes ]; then
+        start_typha
+        rm -f /etc/calico/confd/config/*
+    fi
+
+    # Run confd as a background process.
+    echo "Running confd as background process"
+    BGP_LOGSEVERITYSCREEN="debug" confd -confdir=/etc/calico/confd >$LOGPATH/logd1 2>&1 &
+    CONFD_PID=$!
+    echo "Running with PID " $CONFD_PID
+
+    # Create 3 nodes and enable node mesh BGP password
+    $CALICOCTL apply -f - <<EOF
```

Comment:
Any reason you needed to script this here rather than via a new test directory with the resources within? I thought that's how most of the other tests worked.

As an aside, we need to kill this test framework and burn it straight to the ground. It's horrible and personally I think we should start looking at deprecating it, and preventing new tests from being added. 

We should be able to devise something better for new tests.

Reply:
The only reason I wrote out the tests explicitly as opposed to doing it all through the common framework is so that the tests can go through multiple steps sequentially (which is really just trying to tack an e2e onto the existing framework).